### ISSUE TITLE
Unexpected token: _u16

### ISSUE THREAD

**Issue by anicholson** [hanneskaeufler/crytic#28]
Great shard, thanks for writing!

I ran this against my [gopher.cr] library and encountered this error:

```
Unhandled exception: unexpected token: _u16 (Crystal::SyntaxException)
  from /usr/lib/crystal/compiler/crystal/syntax/lexer.cr:2944:9 in 'raise'
  from /usr/lib/crystal/compiler/crystal/syntax/lexer.cr:2943:5 in 'raise'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:5707:9 in 'unexpected_token'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:5703:5 in 'unexpected_token'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:142:13 in 'parse_multi_assign'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:104:9 in 'parse_expressions_internal'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:3961:7 in 'parse_expressions'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:5729:7 in 'parse_class_def'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:1496:5 in 'parse_class_def:is_abstract:doc'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:955:19 in 'parse_atomic_without_location'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:848:16 in 'parse_atomic'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:560:7 in 'parse_atomic_with_method'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:552:9 in 'parse_prefix'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:536:5 in 'parse_pow'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:535:5 in 'parse_mul_or_div'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:501:7 in 'parse_add_or_sub'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:496:5 in 'parse_shift'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:495:5 in 'parse_logical_and'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:494:5 in 'parse_logical_or'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:493:5 in 'parse_cmp'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:492:5 in 'parse_equality'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:491:5 in 'parse_and'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:490:5 in 'parse_or'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:443:7 in 'parse_range'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:416:7 in 'parse_question_colon'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:306:7 in 'parse_op_assign'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:302:5 in 'parse_op_assign'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:238:7 in 'parse_expression'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:115:7 in 'parse_multi_assign'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:91:7 in 'parse_expressions_internal'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:3961:7 in 'parse_expressions'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:5729:7 in 'parse_module_def'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:1015:15 in 'parse_atomic_without_location'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:848:16 in 'parse_atomic'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:560:7 in 'parse_atomic_with_method'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:552:9 in 'parse_prefix'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:536:5 in 'parse_pow'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:535:5 in 'parse_mul_or_div'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:501:7 in 'parse_add_or_sub'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:496:5 in 'parse_shift'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:495:5 in 'parse_logical_and'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:494:5 in 'parse_logical_or'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:493:5 in 'parse_cmp'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:492:5 in 'parse_equality'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:491:5 in 'parse_and'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:490:5 in 'parse_or'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:443:7 in 'parse_range'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:416:7 in 'parse_question_colon'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:306:7 in 'parse_op_assign'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:302:5 in 'parse_op_assign'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:238:7 in 'parse_expression'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:115:7 in 'parse_multi_assign'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:91:7 in 'parse_expressions_internal'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:3961:7 in 'parse_expressions'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:75:21 in 'parse'
  from /usr/lib/crystal/compiler/crystal/syntax/parser.cr:16:7 in 'parse'
  from lib/crytic/src/crytic/mutation/inject_mutated_subject_into_specs.cr:57:9 in 'process'
  from lib/crytic/src/crytic/mutation/inject_mutated_subject_into_specs.cr:134:13 in 'visit'
  from /usr/lib/crystal/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
  from /usr/lib/crystal/compiler/crystal/syntax/ast.cr:169:27 in 'accept_children'
  from /usr/lib/crystal/compiler/crystal/syntax/visitor.cr:28:11 in 'accept'
  from lib/crytic/src/crytic/mutation/inject_mutated_subject_into_specs.cr:58:9 in 'process'
  from lib/crytic/src/crytic/mutation/inject_mutated_subject_into_specs.cr:134:13 in 'visit'
  from /usr/lib/crystal/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
  from /usr/lib/crystal/compiler/crystal/syntax/ast.cr:169:27 in 'accept_children'
  from /usr/lib/crystal/compiler/crystal/syntax/visitor.cr:28:11 in 'accept'
  from lib/crytic/src/crytic/mutation/inject_mutated_subject_into_specs.cr:58:9 in 'process'
  from lib/crytic/src/crytic/mutation/inject_mutated_subject_into_specs.cr:134:13 in 'visit'
  from /usr/lib/crystal/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
  from /usr/lib/crystal/compiler/crystal/syntax/ast.cr:169:27 in 'accept_children'
  from /usr/lib/crystal/compiler/crystal/syntax/visitor.cr:28:11 in 'accept'
  from lib/crytic/src/crytic/mutation/inject_mutated_subject_into_specs.cr:58:9 in 'process'
  from lib/crytic/src/crytic/mutation/inject_mutated_subject_into_specs.cr:68:9 in 'to_covered_source'
  from lib/crytic/src/crytic/mutation/mutation.cr:130:9 in 'mutated_specs_source'
  from lib/crytic/src/crytic/mutation/mutation.cr:73:21 in 'write_full_source_into_tempfile'
  from lib/crytic/src/crytic/mutation/mutation.cr:57:7 in 'run_mutation'
  from lib/crytic/src/crytic/mutation/mutation.cr:24:24 in 'run'
  from lib/crytic/src/crytic/runner.cr:34:18 in 'run'
  from lib/crytic/src/crytic.cr:51:11 in '__crystal_main'
  from /usr/lib/crystal/crystal/main.cr:97:5 in 'main_user_code'
  from /usr/lib/crystal/crystal/main.cr:86:7 in 'main'
  from /usr/lib/crystal/crystal/main.cr:106:3 in 'main'
  from __libc_start_main
  from _start
  from ???
    ✅ NumberLiteralChange at line 9, column 27%     
```

# Crystal information

```$ crystal --version
Crystal 0.27.0 (2018-11-02)

LLVM: 6.0.1
Default target: x86_64-pc-linux-gnu
```

[gopher.cr]: https://github.com/anicholson/gopher.cr

**Comment by hanneskaeufler** [comment]
Thanks for finding this, made sense, I have fixed this and release 3.0.1. Feel free to hammer this against gopher again, I will do so myself locally as well :)

**Comment by hanneskaeufler** [comment]
@anicholson two things came up when running crytic against gopher.cr:

* You are using minitest.cr, not crystals spec library. Although luckily `crystal spec` still works to run your tests, the "fail fast" option of crystal spec was erroring in your code (requir-ing both "spec" and "minitest/autorun" is a no-go :D). I added an option to specify the preablme, which until recently hardcoded the fail-fast option. tl/dr: Use `./bin/crytic --preamble ""` to run if you are using something other then the crystal spec lib.
* fixture files with relative paths are a problem, and I have no solution currently, only a workaround. So to run crytic in gopher.cr, you need to apply something like this locally:

```
diff --git shard.lock shard.lock
index 99a5adc..1021cb8 100644
--- shard.lock
+++ shard.lock
@@ -1,5 +1,9 @@
 version: 1.0
 shards:
+  crytic:
+    github: hanneskaeufler/crytic
+    version: 3.1.1
+
   minitest:
     github: ysbaddaden/minitest.cr
     version: 0.4.2
diff --git shard.yml shard.yml
index 495ccdc..a40740a 100644
--- shard.yml
+++ shard.yml
@@ -23,6 +23,9 @@ development_dependencies:
   minitest:
     github: ysbaddaden/minitest.cr
     version: "~> 0.4.2"
+  crytic:
+    github: hanneskaeufler/crytic
+    version: "~> 3.1.1"
 
 # uncomment this when sentry becomes 0.25-compatible
 #  sentry:
diff --git spec/gopher/resolvers/directory_resolver_spec.cr spec/gopher/resolvers/directory_resolver_spec.cr
index a7d05d0..5a4e548 100644
--- spec/gopher/resolvers/directory_resolver_spec.cr
+++ spec/gopher/resolvers/directory_resolver_spec.cr
@@ -5,7 +5,8 @@ module Gopher
     let(:config) { Config.new listen_host: "nomatter", listen_port: 70_u16, public_host: "localhost", public_port: 70_u16 }
 
     let(:dr) do
-      root_path : String = File.dirname(__FILE__) + "/../../resources/example_directory"
+      root_path : String =  File.expand_path("./spec/resources/example_directory")
+
       DirectoryResolver.new(root_path: root_path, root_selector: "/1files", config: config)
     end
 
@@ -61,7 +62,7 @@ module Gopher
 
         result = dr.resolve(req)
         resource = result.value.as Resource
-        expected_content = File.read(File.dirname(__FILE__) + "/../../resources/example_directory/ipsum.txt")
+        expected_content = File.read(File.expand_path("./spec/resources/example_directory") + "/ipsum.txt")
 
         expect(resource.content.gets_to_end).must_equal(expected_content)
       end
@@ -103,7 +104,7 @@ module Gopher
 
     describe "Requesting a menu item from a submenu" do
       it "retrieves the resource" do
-        expected_content = File.read(File.dirname(__FILE__) + "/../../resources/example_directory/looks_like_a_file_but_is_a.directory/lol.txt")
+        expected_content = File.read(File.expand_path("./spec/resources/example_directory") + "/looks_like_a_file_but_is_a.directory/lol.txt")
         req = RequestBody.new("/1files/1looks_like_a_file_but_is_a.directory/0lol.txt")
 
         result = dr.resolve(req)
```
